src/script/storage/StorageService.ts

```diff
@@ -391,7 +416,14 @@ export class StorageService {
   async update<T = Object>(storeName: string, primaryKey: string, changes: T): Promise<number> {
     try {
       if (this.isTemporaryAndNonPersistant) {
+        const oldRecord = await this.engine.read<unknown>(storeName, primaryKey);
         await this.engine.update(storeName, primaryKey, changes);
+        const newRecord = await this.engine.read<unknown>(storeName, primaryKey);
+
+        this.dbListeners
```

Comment:
Let's try to not have this code duplicated.
Maybe writing a function `warnListeners(store, eventType, oldRecord, newRecord)` could abstract the implementation away.